File: test/end-to-end/StringIndices/String indices.java
//
// Initialization
String haystack = "Hello, GLS!";

// Searching
String needle = "GLS";
int firstIndexOf = haystack.indexOf(needle);
int secondIndexOf = haystack.indexOf(needle, firstIndexOf + needle.length());

// Results
System.out.println(String.format("Found a first result at: %0$d.", firstIndexOf));

if (secondIndexOf != -1) {
    System.out.println(String.format("Found a second result at: %0$d.", secondIndexOf));
}
//

File: test/end-to-end/StringIndices/String indices.ts
//
// Initialization
let haystack: string = "Hello, GLS!";

// Searching
let needle: string = "GLS";
let firstIndexOf: number = haystack.indexOf(needle);
let secondIndexOf: number = haystack.indexOf(needle, firstIndexOf + needle.length);

// Results
console.log(`Found a first result at: ${firstIndexOf}.`);

if (secondIndexOf !== -1) {
    console.log(`Found a second result at: ${secondIndexOf}.`);
}
//
